src/pynwb/file.py

```diff
@@ -669,36 +668,54 @@ def add_stimulus_template(self, timeseries):
         self._add_stimulus_template_internal(timeseries)
         self._update_sweep_table(timeseries)
 
-    @docval({'name': 'data', 'type': (np.ndarray, list, tuple, pd.DataFrame, DynamicTable, NWBContainer, ScratchData),
-             'help': 'the data to add to the scratch space'},
+    @docval({'name': 'data',
+             'type': ('scalar_data', np.ndarray, list, tuple, pd.DataFrame, DynamicTable, NWBContainer, ScratchData),
+             'doc': 'The data to add to the scratch space'},
             {'name': 'name', 'type': str,
-             'help': 'the name of the data. Only used when passing in numpy.ndarray, list, or tuple',
+             'doc': ('The name of the data. Required and used only when passing in numpy.ndarray, list, tuple, '
+                     'or pandas.DataFrame'),
              'default': None},
             {'name': 'notes', 'type': str,
-             'help': 'notes to add to the data. Only used when passing in numpy.ndarray, list, or tuple',
+             'doc': ('Notes to add to the data. Highly recommended and used only when passing in numpy.ndarray, '
+                     'list, or tuple.'),
              'default': None},
             {'name': 'table_description', 'type': str,
-             'help': 'description for the internal DynamicTable used to store a pandas.DataFrame',
-             'default': ''})
+             'doc': ('Description for the internal DynamicTable used to store a pandas.DataFrame. Highly recommended '
```

Comment:
Why is table_description from notes? table_description is supplied when you have a table and notes are supplied otherwise, and it seems they both serve the same purpose. Would it not be simpler to just have a single argument ``description`` instead?

Reply:
Yes, that would be simpler. @ajtritt - since you had initially designed this, what do you think?

Reply:
Yeah, that would be simpler.